common/v2/features/Dashboard/components/ActionTile.tsx

```diff
@@ -1,22 +1,84 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
 import { Button, Typography } from '@mycrypto/ui';
 
 import { Action } from '../types';
 import './ActionTile.scss';
+import { AnalyticsService, ANALYTICS_CATEGORIES } from 'v2/services';
 
 type Props = Action;
 
-export default function ActionTile({ icon, title, description, link }: Props) {
+const trackAction = (analyticsMessage: string) => {
+  AnalyticsService.instance.track(ANALYTICS_CATEGORIES.DASHBOARD, analyticsMessage);
+};
+
+const ActionTile = React.memo(function ActionTileButton({
+  icon,
+  title,
+  description,
+  path,
+  url,
+  tracking
+}: Props) {
+  const [randomWallet, setRandomWallet] = useState();
+  if (Array.isArray(url)) {
```

Comment:
Simply **NO**.
Instead of changing our view to try to match the data object, we prefer to [malaxate](https://en.wiktionary.org/wiki/malaxate) the data.
The tiles are identical between each other (a link containing img and button) and the component should reflect that. 

From what I understand:
1. we want to randomly select between hardware wallet options 🤢
2. so we add the different options to `actions` list.
3. then we iterate on the list and display ActionTile
4. ActionTile (who is already 2 degrees removed from the initial data structure) now has to be aware how to handle 3 different renders! This is additional complexity just creates more confusion and is harmful in the long run because it creates a strong interdependency between actions in Dashboard and ActionTile.

ActionTile should only be concerned the rendering of its own props.
```
  const selectedActions = actions.map((action) => {
      // we only want to transform the hardware action item so skip the others
      if (!Array.isArray(action.url)) { return action }
      // randomly select a hardware wallet to promote
      const randomIdx = Math.floor(Math.random() * Math.floor(walletOptions))
      return {
         ...action,
         url: action.urls[randomIdx],
         tracking: action.tracking[randomIdx],
         icons: action.tracking[randomIdx],
       }
   })

  // now ActionTile always receives the same props!
```

It could even be agnostic of `tracking` and only receive a click handler. But that's another conversation.